lib/fluent/plugin/buffer.rb

```diff
@@ -761,18 +771,41 @@ def write_step_by_step(metadata, data, format, splits_count, &block)
             begin
               while writing_splits_index < splits.size
                 split = splits[writing_splits_index]
+                formatted_split = format ? format.call(split) : split.first
+                if split.size == 1 && original_bytesize == 0
+                  if format == nil && @compress == :gzip
```

Comment:
`@compress != :text` is more robust against future expanding.

Reply:
Thanks, It is reasonable!

Reply:
Fixed.